### PR TITLE
Fix UseBoundBox

### PR DESCRIPTION
**Pull request by ye-luo**
This solve #1563

The new logic is, the bound box is only active when a lattice is specified in the input and only the open directions have bound check applied.

### CODE REVIEW

File: src/Particle/tests/test_distance_table.cpp [hunk 1]
```diff
@@ -410,8 +410,8 @@ TEST_CASE("distance_pbc_z", "[distance_table][xml]")
   REQUIRE(electrons.SameMass);
 
   // calculate particle distances
-  electrons.Lattice.copy(*SimulationCell);
-  ions.Lattice.copy(*SimulationCell); // is this applied in qmcpack executable?
+  electrons.Lattice = *SimulationCell;
+  ions.Lattice = *SimulationCell; // is this applied in qmcpack executable?
```
[the remark this answers is not in the document]
Comment (ye-luo) [reply src/Particle/tests/test_distance_table.cpp]: I cleaned up all the comments around here.

File: src/Particle/ParticleSet.BC.cpp [hunk 2]
```diff
@@ -66,7 +66,7 @@ void ParticleSet::createSK()
   //for(int ig=0; ig<mySpecies.size(); ++ig)
   //  SubPtcl[ig+1]=SubPtcl[ig]+mySpecies(membersize,ig);
 
-  if (UseBoundBox)
+  if (Lattice.is_from_input)
```
[the remark this answers is not in the document]
Comment (ye-luo) [reply src/Particle/ParticleSet.BC.cpp]: Picked explicitly_defined

File: src/Lattice/CrystalLattice.h [hunk 3]
```diff
@@ -224,56 +226,40 @@ struct CrystalLattice: public LRBreakupParameters<T, D>
...
-  }
-
-  /** assignment operator
-   *@param rhs a tensor representing a unit cell
-   */
-  template<typename T1>
-  CrystalLattice<T, D>& operator=(const Tensor<T1, D>& rhs)
-  {
-    R = rhs;
+    using base_t          = LRBreakupParameters<T, D>;
```
[the remark this answers is not in the document]
Comment (ye-luo) [reply src/Lattice/CrystalLattice.h]: Changed based on the rule.

**Comment by ye-luo** [comment]
I like the suggestion. To avoid merging conflict, I'd like to make the change after this PR and #1664 get merged.

**Comment by prckent** [comment]
Test this please

**Pull request merged**